api/error.go

```diff
@@ -11,3 +12,7 @@ type Error struct {
 func (err Error) MarshalJSON() ([]byte, error) {
 	return json.Marshal(err.Error.Error())
 }
+
+func (err Error) UnmarshalJSON(data []byte) error {
+	return fmt.Errorf(string(data))
```

Comment:
Needs to be `func (err *Error) ...` and set `err.Error`, not return it. Returning an error fails the JSON Unmarshalling.

Also, `string(data)` will presumably end up with undesired `"..."` quotes around the string... This needs to use `json.Unmarshal` internally to decode the raw JSON.

Reply:
My bad, will be fixed in following commit